packages/api/tests/utils/ensembleUtils.ts

```diff
@@ -15,16 +15,89 @@
 
 import mocks from '../mock.factories.js';
 import { ENSEMBLE_BRAND } from '../../src/util/ensemble.js';
+import { IrrigationPrescription } from '../../src/util/ensembleService.types.js';
+import TaskModel from '../../src/models/taskModel.js';
+import { AddonPartner, Farm, Location, ManagementPlan } from '../../src/models/types.js';
 
-export interface Farm {
-  farm_id: string;
-}
-
-export const connectFarmToEnsemble = async (farm: Farm) => {
+export const connectFarmToEnsemble = async (farm: Farm, partner?: AddonPartner) => {
   const [farmAddon] = await mocks.farm_addonFactory({
     promisedFarm: Promise.resolve([farm]),
-    promisedPartner: mocks.addon_partnerFactory({ name: ENSEMBLE_BRAND }),
+    promisedPartner: partner
+      ? Promise.resolve([partner])
+      : mocks.addon_partnerFactory({ name: ENSEMBLE_BRAND, id: undefined }),
   });
 
   return { farmAddon };
 };
+
+type fakeIrrigationPrescriptionsProps = {
+  farmId: Farm['farm_id'];
+  prescriptionIds?: IrrigationPrescription['id'][];
+  locationIds?: Location['location_id'][];
+  managementPlanIds?: ManagementPlan['management_plan_id'][];
+  startTime?: string;
+  endTime?: string;
+};
+
+/**
+ * Returns a list of mocked prescriptions based on a specific farm_id.
+ * TODO: refactor once it is no longer used on beta to be tests specific
+ *
+ * @param farm_id - The ID of the farm to retrieve mock data for.
+ * @returns A promise that resolves to formatted irrigation prescription data.
+ */
+export const fakeIrrigationPrescriptions = async ({
```

Comment:
This isn't currently like our other "fakes" I would like to fix that but maybe once it is not doing both beta and tests mocking.

Reply:
I was working on the assumption task completion flow was not yet testable. Let me see what I can remove now that it is merged. Yes the external id on task would have had to be manually inserted in db or just view the null values.

Granted I shouldn't have tried to merge the mocks probably. Let me see what can be improved.

Reply:
I think this should now be addressed. Sorry the tests were bad indeed. I am so used to just doing something to show it kind of works and moving on. I have a lot of trouble with our tests...

The test now gets prescriptions before linking task, then creates a task with it linked, refetches the prescriptions with tasks to show they all succeed.